src/claimtrie.cpp

```diff
@@ -41,6 +41,68 @@ uint256 getValueHash(const COutPoint& outPoint, int nHeightOfLastTakeover)
     return result;
 }
 
+template <typename T>
+bool equals(const T& value, const COutPoint& outPoint)
+{
+    return value.outPoint == outPoint;
+}
+
+template <typename T>
+bool equals(const T& value, const std::string& name, const COutPoint& outPoint)
+{
+    return value.name == name && value.outPoint == outPoint;
+}
+
+template <typename K, typename V>
+bool equals(const std::pair<K, V>& pair, const std::string& name, const COutPoint& outPoint)
+{
+    return pair.first == name && pair.second.outPoint == outPoint;
+}
+
+template <typename T>
+auto findOutPoint(T& cont, const COutPoint& outPoint) -> decltype(cont.begin())
+{
+    using type = typename T::value_type;
+    static_assert(std::is_same<typename std::remove_const<T>::type, std::vector<type>>::value, "T should be a vector type");
+    return std::find_if(cont.begin(), cont.end(), [&outPoint](const type& val) {
+        return equals(val, outPoint);
+    });
+}
+
+template <typename T>
+auto findOutPoint(T& cont, const std::string& name, const COutPoint& outPoint) -> decltype(cont.begin())
```

Comment:
`-> decltype` is c++14, no?

Reply:
No, https://en.cppreference.com/w/cpp/language/decltype